python/pythonlab/neighborhood/neighborhood/painter.py

```diff
@@ -16,28 +18,41 @@ def __init__(self, x=0, y=0, direction="East", paint=0):
     """
     self.x = x
     self.y = y
-    self.direction = direction
+    self.direction = Direction(direction)
     self.remaining_paint = paint
     # Create a reference to the world singleton
     self.world = World()
     # If the grid is not set, set it from the default file
     if (self.world.grid is None):
       self.world.set_grid_from_file()
+    Painter.last_id += 1
+    self.id = f"painter-{Painter.last_id}"
+    self.send_initialization_message(x, y, direction, paint)
 
+  def _send_signal(self, key, detail=None):
+      """
+      Helper method to create and print a signal message.
 
+      Args:
+        key (str): The key for the signal message (e.g., NeighborhoodSignalKey).
+        detail (dict): Optional additional details for the signal message.
+      """
+      detail = detail or {}
+      signal_message = NeighborhoodSignalMessage(SignalMessageType.PAINTER, key, detail)
+      print(signal_message.get_formatted_message())
+      
   def turn_left(self):
     """
     Turn the painter one compass direction left (i.e. North -> West).
     """
-    signal_message = NeighborhoodSignalMessage(SignalMessageType.PAINTER, NeighborhoodSignalKey.TURN_LEFT.value)
-    print(signal_message.get_formatted_message())
+    self.direction.turn_left()
+    self._send_signal(NeighborhoodSignalKey.TURN_LEFT.value, {'id': self.id, 'direction': self.direction.value})
```

Comment:
Should the helper method provide the id to the `detail` object so we don't have to pass it every time?